packages/apollo-datasource-rest/src/RESTDataSource.ts

```diff
@@ -8,7 +8,15 @@ export abstract class RESTDataSource<TContext = any> {
   httpCache!: HTTPCache;
   context!: TContext;
 
-  willSendRequest?(request: Request): void;
+  public willSendRequest?(request: Request): void;
+
+  public willReceiveCache(httpCache: HTTPCache) {
```

Comment:
Hmmm, not sure I understand the benefits of this. It will actually open up users to mistakes when they override these methods and forget to call super (or set the properties themselves).

If the idea is that we'd like to have a lifecycle method that's called when the datasource is fully initialized, we should probably define it separately and call it when both cache and context have been set.

Reply:
That makes sense! We don't want users to mess with those "lifecycles", then we should probably have a `ready` or `willReceiveRequest` lifecycle for after the httpCache and context are set.